### DISCUSSION TITLE
netbox-ui-plugin issues (pull access denied and wrong environment)

### DISCUSSION THREAD

**Discussion by cjbody** [netbox-community/netbox-docker#649] [Getting Help] [answered]
Fairly new to linux and docker stuff so apologies if this is missing info or anything.

I'm running a docker-compose for Netbox using this image https://github.com/netbox-community/netbox-docker
I'm trying to get the nextbox-ui-plugin using https://github.com/iDebugAll/nextbox-ui-plugin

I encounter several issues, the primary is the custom image can't be pulled after changing the compose image name to `image: netbox-custom:latest` which is built (I assume) from the command, `$ docker build -t netbox-custom .`. The error I get when trying to docker-compose up is this:

root@Netbox:~/netbox-docker# docker-compose up
Pulling netbox-worker (netbox-custom:latest)...
ERROR: pull access denied for netbox-custom, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

The next issue I get when I change the image name back to `netboxcommunity/netbox:latest` but still have the `PLUGINS = ['nextbox_ui_plugin']` in the configuration.py file. This plugin error about the wrong environment which I'm confused about since this is a docker container.

netbox-housekeeping_1  | django.core.exceptions.ImproperlyConfigured: Unable to import plugin nextbox_ui_plugin: Module not found. Check that the plugin module has been installed within the correct Python environment.
netbox_1               | django.core.exceptions.ImproperlyConfigured: Unable to import plugin nextbox_ui_plugin: Module not found. Check that the plugin module has been installed within the correct Python environment.
netbox_1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox_1               | ⏳ Waiting on DB... (0s / 30s)

**Comment by cjbody** [comment, marked as answer]
I figured it out - a few things were both blocked on the router side (allowed everything that popped up for the installation during the docker build -t netbox-custom . command. I also had to alter that command to: $ DOCKER_BUILDKIT=1 docker build -t netbox-custom .

**Reply by Kniickk** [reply]
Hello cjbody,

could you explain your Solution a bit more? 
I have the Same problem but i dont understand your solution for it.
Especially the part with DOCKER_BUILDKIT=1.



ERROR: The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.

I get the following errors while building the Container with all the config changes.
```
Continue with the new image? [yN]y
Pulling netbox-housekeeping (netbox-custom:latest)...
ERROR: pull access denied for netbox-custom, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```
I would be thankfull if i could resolve the Issue with your help.

**Reply by Kniickk** [reply]
So we fixed it.
The Docker command that we used was 
DOCKER_BUILDKIT=1 docker build -t netbox-custom .
The process that was startet ran than into an error with no connection.
We have a proxy and the install is made with pip3 in the Dockerfile from Nextbox (Netbox/nextbox-ui-plugin). Normaly Pip should use the Export proxy, in ouer case it dident and was blocked by ouer firewalls.
We resolved the issue by edditing the Pip3 command as followed:

Original: RUN pip3 install  --no-cache-dir nextbox-ui-plugin
Modified: RUN pip3 install --proxy X.X.X.X:X --no-cache-dir nextbox-ui-plugin

For the correct way to defin your Proxy, look up the Documentation for pip3
Also we needed to enable docker you can see how to do that here:
https://docs.docker.com/develop/develop-images/build_enhancements/#to-enable-buildkit-builds

Hope this can help anyone
